akka-stream/src/main/scala/akka/stream/scaladsl/FlowWithContext.scala

```diff
@@ -16,50 +16,158 @@ import akka.stream._
+   * context propagation here.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.via]]
+   */
   def via[Ctx2, Out2, Mat2](flow: Graph[FlowShape[(Out, Ctx), (Out2, Ctx2)], Mat2]): Repr[Ctx2, Out2]
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.map]].
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.map]]
+   */
   def map[Out2](f: Out ⇒ Out2): Repr[Ctx, Out2] =
     via(flow.map { case (e, ctx) ⇒ (f(e), ctx) })
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.mapAsync]].
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.mapAsync]]
+   */
   def mapAsync[Out2](parallelism: Int)(f: Out ⇒ Future[Out2]): Repr[Ctx, Out2] =
     via(flow.mapAsync(parallelism) { case (e, ctx) ⇒ f(e).map(o ⇒ (o, ctx))(ExecutionContexts.sameThreadExecutionContext) })
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.collect]].
+   *
+   * Note, that the context of elements that are filtered out is skipped as well.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.collect]]
+   */
   def collect[Out2](f: PartialFunction[Out, Out2]): Repr[Ctx, Out2] =
     via(flow.collect {
       case (e, ctx) if f.isDefinedAt(e) ⇒ (f(e), ctx)
     })
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.filter]].
+   *
+   * Note, that the context of elements that are filtered out is skipped as well.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.filter]]
+   */
   def filter(pred: Out ⇒ Boolean): Repr[Ctx, Out] =
     collect { case e if pred(e) ⇒ e }
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.filterNot]].
+   *
+   * Note, that the context of elements that are filtered out is skipped as well.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.filterNot]]
+   */
   def filterNot(pred: Out ⇒ Boolean): Repr[Ctx, Out] =
     collect { case e if !pred(e) ⇒ e }
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.grouped]].
+   *
+   * Each output group will be associated with a `Seq` of corresponding context elements.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.grouped]]
+   */
   def grouped(n: Int): Repr[immutable.Seq[Ctx], immutable.Seq[Out]] =
     via(flow.grouped(n).map { elsWithContext ⇒
       val (els, ctxs) = elsWithContext.unzip
       (els, ctxs)
     })
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.sliding]].
+   *
+   * Each output group will be associated with a `Seq` of corresponding context elements.
+   *
+   * @see [[akka.stream.scaladsl.FlowOps.sliding]]
+   */
   def sliding(n: Int, step: Int = 1): Repr[immutable.Seq[Ctx], immutable.Seq[Out]] =
     via(flow.sliding(n, step).map { elsWithContext ⇒
       val (els, ctxs) = elsWithContext.unzip
       (els, ctxs)
     })
 
+  /**
+   * Context-preserving variant of [[akka.stream.scaladsl.FlowOps.mapConcat]].
+   *
+   * The context of the input element will be associated with each of output elements calculated from
```

Comment:
each of output elements -> each output element?

Reply:
Thanks.